src/choi_operators.jl

```diff
@@ -0,0 +1,141 @@
+export is_choi_matrix, choi_matrix, ChoiOp, choi_evolve
+
+"""
+    is_choi_matrix(
+        Λ :: AbstractMatrix,
+        dims :: Vector{Int};
+        atol=ATOL :: Float
+    ) :: Bool
+
+Returns `true` if matrix `Λ` is a Choi operator representation of a quantum
+channel.
+
+The requirements on `Λ` are ([https://arxiv.org/abs/1111.6950](https://arxiv.org/abs/1111.6950)):
+* Is Hermitian, ``\\Lambda = \\Lambda^{\\dagger}``.
+* Is trace-preserving, ``\\text{Tr}_A[\\Lambda^{AB}]=\\mathbb{I}_B``.
+* Is completely-positive, ``\\Lambda \\geq 0``.
+"""
+function is_choi_matrix(
+    Λ :: AbstractMatrix,
+    dims :: Vector{Int};
+    atol=ATOL::Float64
+) :: Bool
+    if !is_hermitian(Λ, atol=atol)
+        @warn "The Choi matrix `Λ` is not Hermitian-preserving.`"
+        return false
+    elseif !is_positive_semidefinite(Λ, atol=atol)
+        @warn "The Choi matrix `Λ` is not completely-positive."
+        return false
+    elseif !isapprox(partial_trace(Λ, dims, 1), I, atol=atol)
+        @warn "The Choi matrix `Λ` is not trace-preserving."
+        return false
+    end
+
+    return true
+end
+
+"""
+    choi_matrix(𝒩 :: Function, dims :: Vector{Int}) :: Matrix{ComplexF64}
+
+Returns the Choi operator of a channel. The Choi matrix is constructed as
+
+```math
+        \\Lambda^{AB} = \\sum_{i,j \\in [d_A]} E^A_{i,j} \\otimes \\mathcal{N}^B(E_{i,j}) ,
```

Comment:
If this is already your standard then that's obviously fine, but I think the superscript B on \\mathcal{N} is sort of weird. I guess my suggestion would be
```suggestion
        \\Lambda^{AB} = \\sum_{i,j \\in [d_A]} E^A_{i,j} \\otimes \\mathcal{N}^{A' \\to B}(E_{i,j}^{A'}) ,
```
or something similar, but I grant it will be less elegant.